code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -794,18 +900,29 @@ var/global/list/damage_icon_parts = list()
 			bloodsies.color = wear_mask.blood_color
 			standing.overlays	+= bloodsies
 		overlays_standing[FACEMASK_LAYER]	= standing
-	else
-		overlays_standing[FACEMASK_LAYER]	= null
 	if(update_icons)   update_icons()
 
 
 /mob/living/carbon/human/update_inv_back(var/update_icons=1)
+
+	overlays_standing[BACK_LAYER] = null
 	if(back)
 		back.screen_loc = ui_back	//TODO
 
 		//determine the icon to use
 		var/icon/overlay_icon
-		if(back.icon_override)
+		var/overlay_state = ""
+
+		if(back.contained_sprite)
+			if (back.species_tag)
+				overlay_state += "[back.species_tag]_"
+			overlay_state += "[back.item_state][WORN_BACK]"
+
+			if(back.icon_override)
+				overlay_icon = back.icon_override
+			else
+				overlay_icon = back.icon
+		else if(back.icon_override)
 			overlay_icon = back.icon_override
 		else if(istype(back, /obj/item/weapon/rig))

```

Comment:
I fail to see how, it should work just fine

Are you talking about only the module, or the full suit? The individual pieces have their own icon information anyway. But it should be very possible to put the module and all pieces in both open and closed states, into one dmi file. I'm instructing datberry in doing exactly that with his civilian rig, ill report if any problems arise